Dev linux improvements

alexankitty opened the pull request:
* Add automatic installing of proton dependencies (vc_redist and dotnetdesktop)
* Make how to launch show by default for linux after installation to provide commands more readily to the user
* Add a button to open the steam game properties dialogue directly
* Add various linux utilities to make it easy to install dependencies
* Supports dependency installation via native and flatpak protontricks with native being preferred.

Let me know if any additional changes need to be made. This should hopefully reduce the amount of effort and confusion people have with getting MelonLoader running under linux.

slxdy commented:
Sounds cool. The only issue I have with this is the dotnet installer. This is unnecessary as ML 0.7 will do this by itself (the linux implementation for this is still needed however, but works fine under Proton)

slxdy commented:
Or, well actually, it's both dependencies, because they won't be needed for ML 0.7. Requiring the user to have Protontricks installed is pretty unnecessary in that case

slxdy commented:
Also please follow the default coding conventions. I see a lot of nests being opened on the same line, and no blank lines after nests.

```cs
if (abc)
    return;
if (def){
    //stuff
}
Ghi();
```

Should become:
```cs
if (abc)
    return;

if (def)
{
    //stuff
}

Ghi();
```

Same applies to new function nests

alexankitty commented:
So wait, if the dependencies are unnecessary then should I just drop it down to the game properties button and leave it at that?

slxdy commented:
Yea, unfortunately :/
I appreciate the effort, but unfortunately it's not needed anymore

alexankitty commented:
It's fine, I understand. It was still a good learning experience for me so the effort isn't entirely wasted (and I kinda got a game for free out of it cause someone wanted it to be a little easier to use melonloader). My bad for not consulting the main codebase for this information. With that, the steam fame properties I'm using is platform agnostic and doesn't have to be a Linux Util, that's just where I put it with everything else Linux related I was working on. I think this could be implemented as a button on the main page as well if windows users want to make use of it, but let me know what you think about that. I'd also want to move the method to a different class, would MLManager be better for this?

alexankitty commented:
Removed all the extra code, and cleaned up some code style issues I could find.
Let me know if that's sufficient, and if you want me to adjust the game properties button at all.

slxdy commented on MelonLoader.Installer/MLManager.cs:
```diff
@@ -332,7 +346,7 @@ public static void SetLocalZip(string zipPath, InstallProgressEventHandler? onPr
     }
 
-    public static async Task InstallAsync(string gameDir, bool removeUserFiles, MLVersion version, bool linux, bool x86, InstallProgressEventHandler? onProgress, InstallFinishedEventHandler? onFinished)
+    public static async Task InstallAsync(string gameDir, string? id, bool removeUserFiles, MLVersion version, bool linux, bool x86, InstallProgressEventHandler? onProgress, InstallFinishedEventHandler? onFinished)
```
Please rename the id parameter and specify what it's for (ex. `gameId`)

slxdy commented on MelonLoader.Installer/Views/DetailsView.axaml.cs:
```diff
@@ -138,12 +138,27 @@ private void InstallHandler(object sender, RoutedEventArgs args)
             (errorMessage) => Dispatcher.UIThread.Post(() => OnOperationFinished(errorMessage)));
     }
 
+    private void GamePropsHandler(object sender, RoutedEventArgs args)
```
Keep in mind that the installer supports multiple launchers. Make sure you only show this on Steam games or add support for other games

slxdy commented on MelonLoader.Installer/MLManager.cs:
```diff
@@ -49,9 +50,22 @@ public static async Task<bool> Init()
         return inited;
     }
 
+    public static void OpenSteamGameProperties(string? appId)
```
This class is meant for MelonLoader-related operations only. Move this to the SteamLauncher class

slxdy commented on MelonLoader.Installer/MLManager.cs:
```diff
@@ -49,9 +50,22 @@ public static async Task<bool> Init()
+            return;
+        }
+        
+        Process.Start(new ProcessStartInfo(){
```
Please abide by the default coding conventions; fix the nest format

slxdy replied:
Add a try-catch block in case it fails to open for whatever reason

slxdy commented:
Also, this PR should be retargeted towards the `development` branch

slxdy commented:
I will check the UI changes once you fix the points above and get approved for workflow checks

alexankitty commented:
It probably won't be until tomorrow when I get this done. Today has been exhausting at work.
